src/datepicker/__tests__/datepicker.test.js

```diff
@@ -229,7 +229,7 @@ describe('Datepicker', () => {
     });
   });
 
-  test('fires the onchange only when date length is same as formatString with single date', () => {
+  test('does not fire the onchange when date length is less than passed formatString with single date', () => {
```

Comment:
I hope it's okay I changed this a bit to match the implementation because the guard is (a < b) not (a !== b)